pkg/destroy/azure/azure.go

```diff
@@ -488,6 +482,13 @@ func deleteResourceGroup(ctx context.Context, client resources.GroupsClient, log
 	return nil
 }
 
+func isNotFoundError(err error) bool {
+	if detailedErr, ok := err.(autorest.DetailedError); ok && detailedErr.StatusCode == http.StatusNotFound {
```

Comment:
I noticed the other methods use `errors.As()` rather than casting like this. Then again, they also use hardcoded response codes rather than consts. :man_shrugging:

Reply:
Let's stick with `errors.As`. That will accept a DetailedError which is wrapped in another error.